### PR TITLE
TINY-6242: Fixed an issue when resizing used fixed table sizing mode

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6242

Description of Changes:
* Fixed an issue I originally missed that got picked up during QA. The issue was that we were syncing the cell sizes before doing the column resize, which meant the column resize calculations were incorrect.

Pre-checks:
* [x] ~Changelog entry added~ (Fix for an unreleased change that already has a changelog)
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts [hunk 1]
```diff
@@ -71,6 +64,17 @@ export const getResizeHandler = function (editor: Editor): ResizeHandler {
       // For preserve table we want to always resize the entire table. So pretend the last column is being resized
       const col = Settings.isPreserveTableColumnResizing(editor) || isRightEdgeResize ? getNumColumns(table) - 1 : 0;
       Adjustments.adjustWidth(table, width - startW, col, resizing, tableSize);
+    // Handle the edge case where someone might fire this event without resizing.
+    // If so then we need to ensure the table is still using percent
+    } else if (Util.isPercentage(startRawW)) {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: I was talking to @ltrouton about this and we were wondering if we should just remove this block? As core will never fire the resize event if the sizes hasn't changed, so this is a very large edge case.

What do others think?

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts (hunk 1 above)
Comment (ashfordneil) [reply modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: I'm like a 4/5 for agreeing that you should remove it, but I'm only 1/5 confident.

**Review by ashfordneil — APPROVED** [review]

File: modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts (hunk 1 above)
Comment (metricjs) [reply modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: Which event is this referring to? Just wondering how likely it is that an integrator will fire it....

**Review by metricjs — APPROVED** [review]

File: modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: The `ObjectResized` event, specifically for tables.

Comment (metricjs) [reply modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: ...I'm leaning towards leaving it in then, just in case. We know some people hook into that event, so even though it's an edge case, since we already have code for it it seems worth dealing with?

File: modules/tinymce/src/plugins/table/test/ts/browser/ResizeTableTest.ts [hunk 2]
```diff
@@ -298,11 +310,14 @@ UnitTest.asynctest('browser.tinymce.plugins.table.ResizeTableTest', (success, fa
...
-      NamedChain.direct('widths', Chain.mapper((widths: any) => widths.element), 'element'),
-      NamedChain.write('lastColWidth', TableTestUtils.cGetCellWidth(0, 1)),
-      NamedChain.read('lastColWidth', Chain.op((width: any) => {
-        Assertions.assertEq(`Last column raw width ${width.raw + width.unit} should be ~106px`, true, Math.abs(106 - width.raw) < 3);
-        Assertions.assertEq('Last column unit width', 'px', width.unit);
+      NamedChain.read('widths', Chain.op((widths: any) => {
+        const firstColWidth = widths.colWidthsAfter[0];
+        const lastColWidth = widths.colWidthsAfter[1];
+        // Note: Use 96px as the padding + borders are about 14px which adds up to ~110px per cell
+        Assertions.assertEq(`First column raw width ${firstColWidth.raw + firstColWidth.unit} should be ~96px`, true, Math.abs(96 - firstColWidth.raw) < 3);
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/plugins/table/test/ts/browser/ResizeTableTest.ts]: Fixed, though going to log a tech debt for extracting the approx assertion. Ideally, that should end up in bedrock or agar.

File: modules/tinymce/src/plugins/table/test/ts/browser/ResizeTableTest.ts [hunk 3]
```diff
@@ -325,11 +340,45 @@ UnitTest.asynctest('browser.tinymce.plugins.table.ResizeTableTest', (success, fa
...
+      cClearResizeEventData,
+      NamedChain.read('editor', ApiChains.cSetContent('')),
+      NamedChain.direct('editor', cInsertResizeMeasure(TableTestUtils.cDragHandle('ne', -250, 0), TableTestUtils.cInsertRaw(percentTable)), 'widths'),
+      NamedChain.read('widths', cAssertWidthAfterResize(53, true)),
+      NamedChain.read('widths', Chain.op((widths: any) => {
+        const firstColWidth = widths.colWidthsAfter[0];
+        const lastColWidth = widths.colWidthsAfter[1];
+        Assertions.assertEq(`First column raw width ${firstColWidth.raw + firstColWidth.unit} should be ~95%`, true, Math.abs(95 - firstColWidth.raw) < 3);
+        Assertions.assertEq('First column unit width', '%', firstColWidth.unit);
+        Assertions.assertEq(`Last column raw width ${lastColWidth.raw + lastColWidth.unit} should be ~5%`, true, Math.abs(5 - lastColWidth.raw) < 3);
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/plugins/table/test/ts/browser/ResizeTableTest.ts]: Copy/paste fail, so thanks for the catch!

File: modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/plugins/table/main/ts/actions/ResizeHandler.ts:69]: I'll leave it as is then.
